### ISSUE TITLE
DisplayAlert Cancel disables ToolbarItems

### ISSUE THREAD

**Issue by Mephisztoe** [dotnet/maui#5723]
### Description

When using a ToolBarItem for a command and have this command ask for user feedback first, then cancelling this command results in the ToolBarItem being disabled.

It seems as if the DisplayAlert code not only puts a gray layer above everything else to make it appear modal but also disables everything... but "forgets" resetting the controls' state back to whatever it was before (enabled for instance).

![image](https://user-images.githubusercontent.com/5309613/161164136-16df8d0a-0a61-495f-bf8b-40db0db8392e.png)

![image](https://user-images.githubusercontent.com/5309613/161164171-160ff5a5-77af-4831-9c30-74efff36ceae.png)

![image](https://user-images.githubusercontent.com/5309613/161164192-172c0532-57a7-4242-aae5-72315f808ec8.png)

The attached code seems slightly over-engineered for reproducing this issue. It's a stripped down version of some real apps' code.

[MauiIssueToolbarItemDisabled.zip](https://github.com/dotnet/maui/files/8393580/MauiIssueToolbarItemDisabled.zip)

### Steps to Reproduce

1. Add a ToolBarItem to a View
2. Bind the ToolBarItem to a Command
3. Have the Command ask for user feedback using DisplayAlert
4. Cancel the dialog

Now the ToolBarItem is disabled.

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android

### Did you find any workaround?

When navigating to another view and back to the view containing the ToolBarItem, the Item is enabled again - of course.

**Comment by kristinx0211** [comment]
verified repro on Android 11.

**Comment by Mephisztoe** [comment]
After some researching I found out what's the cause of this, although I cannot exactly explain why.

Before switching to the MAUI Community Toolkit for its awesome MVVM implementation, I used Refractored.MvvmHelpers. 
The Send Command was implemented then like this:

```csharp
public ICommand SendCommand => new AsyncCommand(async () =>
{
    bool answer = await App.Current.MainPage.DisplayAlert("Confirmation", "The provided information is correct and ready for transmission.", "Confirm", "Cancel");
    if (answer == true)
    {
        await Shell.Current.Navigation.PopAsync();
    }
});
```

I used an AsyncCommand because DisplayAlert demands being called asynchronously (i.e. it returns Task<bool>). 

**This** works.

Then I switched to MCT and implemented the Send Command like this:

```csharp
[ICommand]
async Task Send()
{
    bool answer = await Shell.Current.DisplayAlert("Confirmation", "The provided information is correct and ready for transmission.", "Confirm", "Cancel");
    if (answer == true)
    {
        // Some other stuff happens here
        await Shell.Current.Navigation.PopAsync();
    }
}
```

The SourceGenerator translates this to an AsyncRelayCommand. 

And this is when things break apart...

I found out about this when trying to replace the DisplayAlert method call with the freshly included ShowPopup() method within the recent release of the MAUI Community Toolkit.

As long as I call it synchronously it works like a charm.

For example (for the sake of this example, no return processing included):
```csharp
[ICommand]
void Send()
{
    App.Current.MainPage.ShowPopup(new ConfirmationPopup());
}
```

But when I do it asynchronously, it behaves like DisplayAlert, disabled the ToolbarItem and does not re-enable it again:

```csharp
[ICommand]
async Task Send()
{
    await App.Current.MainPage.ShowPopupAsync(new ConfirmationPopup());
}
```

In Germany we say "Vielleicht habe ich ein Brett vorm Kopf". ;-)
Can anyone explain this to me? 

Since DisplayAlert is somewhat limited in regards to its styling options, I guess I will switch to Popups anyway. However, I need to understand what's happening here nonetheless so that I don't walk into any other traps like this in the future...
